src/core/net/srp_client.cpp

```diff
@@ -2066,6 +2075,13 @@ void Client::ProcessAutoStart(void)
         if ((mAutoStart.GetState() != AutoStart::kSelectedAnycast) ||
             (mAutoStart.GetAnycastSeqNum() != anycastInfo.mSequenceNumber))
         {
+            // If anycast sequence number changes when the client is already running or if
+            // anycast sequence number is already initialized, then use higher delays as all the clients
+            // might try to register around the same time
+            if (IsRunning() || (0 != mAutoStart.GetAnycastSeqNum()))
+            {
+                mShouldUseLongDelays = true;
+            }
```

Comment:
Do we need this check? Not sure if we need this block and may be redundant? I think it will already covered by the new check added below and the fact that `shouldRestart` is set here?

Reply:
**shouldRestart** would be set, even for the scenario where SRP client comes up for the first time (which could be like device pairing scenario, where only this client would register to the SRP server. In those cases, I am thinking we should not use long delays, as it can impact the overall pairing latency. 

 IsRunning ensures that we apply the long delays only when the client is already registered and there is a sequence number change from a previously registered value?

Reply:
> IsRunning ensures that we apply the long delays only when the client is already registered and there is a sequence number change from a previously registered value?

I still think we don't this check here. 

Since `IsRunning()` is already checked in the code block where `mUseUpdateTxDelayLong` is potentially set. So the long delay will be applied under the following conditions:
- The client is running (as guaranteed by the existing `IsRunning()` check).
- There is a change in either the server's address, or anycast's sequence number.

```c++
    if (IsRunning())
    {
        VerifyOrExit((GetServerAddress() != serverSockAddr) || shouldRestart);
        // Use long delays when the SRP client is already running,
        // and there is a change in SRP server address
        mUseUpdateTxDelayLong = true;
        Stop(kRequesterAuto, kResetRetryInterval);
    }
```